libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2034,6 +2036,19 @@ void TypeChecker::endVisit(Literal const& _literal)
 				"For more information please see https://solidity.readthedocs.io/en/develop/types.html#address-literals"
 			);
 	}
+	if (_literal.isHexNumber() && _literal.subDenomination() != Literal::SubDenomination::None)
+	{
+		if (v050)
+			m_errorReporter.fatalTypeError(
+				_literal.location(),
+				"Hexadecimal numbers cannot be used with unit denominations."
```

Comment:
We could suggest to use `0x1234 * 1 day` if they really mean it.

Reply:
Please include that to avoid user complaints 😉

Reply:
OK, see commit https://github.com/ethereum/solidity/pull/3628/commits/454b470ceed799cb22ceaef6ea2b0bbea7165df9.